34th-Sopt-Assignment/34th-Sopt-Assignment/Main/views/HomeViewController.swift

```diff
@@ -0,0 +1,220 @@
+//
+//  MainViewController.swift
+//  34th-Sopt-Assignment
+//
+//  Created by 김민서 on 4/24/24.
+//
+
+import Foundation
+import UIKit
+import Then
+import SnapKit
+
+final class HomeViewController: UIViewController, UICollectionViewDelegate {
+  
+    private lazy var carouselViewController: CarouselViewController = {
+        let viewController = CarouselViewController()
+        return viewController
+    }()
+
+    //화면 전체 스크롤
+    private let scrollView = UIScrollView()
+    private var contentView = UIView()
+    
+    private let tvingLogo = UIImageView()
+    private let airplay = UIButton()
+    private let profile = UIImageView()
+    private let contentLabel = UILabel()
+    private let popularLabel = UILabel()
+    private let showAll = UILabel()
+    private let arrow = UIButton()
+    
+
+    
+    //콜렉션 뷰
+    private let contentCollectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.backgroundColor = .clear // 콜렉션 뷰 배경색을 투명하게 설정
+        return collectionView
+    }()
+    
+
+    private var itemData = ContentModel.dummy()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+       
+        setStyle()
+        setLayout()
+        register()
+        setDelegate()
+    }
+    
+    private func setLayout() {
+     
+       
+        self.view.backgroundColor = .black
+        self.view.addSubview(scrollView)
+
+  
+        [contentView, tvingLogo, profile,airplay].forEach {
+            scrollView.addSubview($0)
+        }
+
+        
+        [carouselViewController.view,contentLabel,popularLabel, showAll, arrow, contentCollectionView].forEach {
+            contentView.addSubview($0)
+        }
+   
+        
+        carouselViewController.view.snp.makeConstraints {
+            $0.top.equalToSuperview()
+            $0.leading.trailing.equalToSuperview()
+            $0.height.equalTo(500)
+        }
+        contentCollectionView.snp.makeConstraints {
+            $0.top.equalTo(contentLabel.snp.bottom).offset(5)
+            $0.leading.equalToSuperview().inset(15)
+            $0.trailing.equalToSuperview()
+            $0.height.equalTo(163)
+        }
+        
+   
+        
+        scrollView.snp.makeConstraints {
+            $0.top.equalToSuperview().inset(-50)
```

Comment:
-50으로 하신 이유가 궁금합니다! 만약 위에 노치 부분때문에 내려주신거면 
```suggestion
            $0.top.equalTo(view.safeAreaLayoutGuide.snp.top)
```
이런 식으로도 사용 가능하고, 위에 상단바가 있어서 그런경우엔 
```suggestion
            $0.top.equalTo(위에 위치한 컴포넌트.snp.bottom)
```
이런식으로도 가능합니다!

Reply:
헉 이거 때문에 야매로 inset줬는데 이런 해결책이,, 감삼당~!